### PR TITLE
Add/edit host view: add fields for the connection-related preferences.

### PR DESCRIPTION
**Pull request by khorimoto**
This PR adds all the remaining UI fields and hooks them up to the fragment's HostBean object.

Remaining work includes:
  1) Passing in initial data to the fragment if it exists.
  2) Saving the host after it has been edited.
  3) UI polish (pending specs from @bengolds). The current design is very raw/ugly.

### CODE REVIEW

**Comment by jklein24** [comment]
Can you send a screenshot and fix those lint errors?

**Comment by khorimoto** [comment]
Fixed lint errors. Screenshot below.

screenshot from 2015-10-05 16 47 40

File: app/src/main/java/org/connectbot/HostEditorFragment.java [hunk 1]
```diff
@@ -128,12 +148,16 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
 		ArrayAdapter<String> transportSelection = new ArrayAdapter<>(
-				getActivity(),
-				android.R.layout.simple_spinner_item,
-				TransportFactory.getTransportNames());
+				getActivity(), android.R.layout.simple_spinner_item, transportNames);
 		transportSelection.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
 		mTransportSpinner.setAdapter(transportSelection);
+		for (int i = 0; i < transportNames.length; i++) {
+			if (transportNames.equals(mHost.getProtocol())) {
+				mTransportSpinner.setSelection(i);
```
Comment (jklein24) [inline app/src/main/java/org/connectbot/HostEditorFragment.java]: break;? Should be consistent one way or the other.

Comment (khorimoto) [inline app/src/main/java/org/connectbot/HostEditorFragment.java]: Done.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -596,5 +596,6 @@
 	<string name="protocol_spinner_label">Protocol</string>
 	<!-- Label for button which expands/collapses section. -->
 	<string name="expand">Expand</string>
+	<string name="hostpref_authagent_with_confirmation">require confirmation</string>
```
Comment (jklein24) [inline app/src/main/res/values/strings.xml]: Can you add a comment above for context?

**Comment by khorimoto** [comment]
Added string comment.

**Pull request merged**